blog/2025-11-12-best-practises-for-writing-promises/index.mdx

```diff
@@ -0,0 +1,559 @@
+```
+
+We can now use the kratix cli to ensure these dependencies are installed when the Promise’s Configure Pipeline runs with the
+`update dependencies` command:
+
+```bash
+kratix update dependencies dependencies --image ghcr.io/syntasso/redis-dependencies:v0.0.2
+```
+
+This command:
+
+- Places the dependencies in the `workflows/promise/configure/dependencies/configure-deps/resources` directory
+- Creates a Dockerfile that copies the contents of the resource directory into the `/kratix/outputs` directory so they will be
+  written to all appropriate destinations
+- Adds the `promise.configure` step to the `pipeline.yaml`
+
+Your directory should now look something like this:
+
+```bash
+├── README.md
+├── dependencies
+│   ├── databases.spotahome.com_redisfailovers.yaml
+│   └── operator-bundle.yaml
+├── example-resource.yaml
+├── promise.yaml
+└── workflows
+    └── promise
+        └── configure
+            └── dependencies
+                └── configure-deps
+                    ├── Dockerfile
+                    ├── resources
+                    │   ├── databases.spotahome.com_redisfailovers.yaml
+                    │   └── operator-bundle.yaml
+                    └── scripts
+                        └── pipeline.sh
+
+9 directories, 9 files
+```
+
+And your promise.yaml should include the below:
+
+```yaml
+  workflows:
+    config: {}
+#highlight-start
+    promise:
+      configure:
+      - apiVersion: platform.kratix.io/v1alpha1
+        kind: Pipeline
+        metadata:
+          name: dependencies
+        spec:
+          containers:
+          - image: ghcr.io/syntasso/redis-dependencies:v0.0.2
+            name: configure-deps
+#highlight-end
+    resource: {}
+```
+
+Now it is time to build the image that will run as part of the  `promise.configure` step. If you are running your cluster on kind,
+you can build and load the image to your cluster with:
+
+```bash
+docker build --tag ghcr.io/syntasso/redis-dependencies:v0.0.1 workflows/promise/configure/dependencies/configure-deps
+
+kind load docker-image ghcr.io/syntasso/redis-dependencies:v0.0.1 --name platform
+```
+
+We can now install this early iteration of the Promise!
+
+```bash
+kubectl apply --filename promise.yaml
+```
+
+You will soon see the Redis dependencies appearing on your worker cluster. Watch the the operator pod as it is created with:
+
+```bash
+kubectl get pods --context kind-worker --watch
+```
+
+You should eventually see output similar to the following:
+
+```bash
+NAME                             READY   STATUS    RESTARTS   AGE
+redisoperator-7f584c6969-zc8sx   1/1     Running   0          41s
+```
+
+You now have a Promise that installs the Redis operator and its its dependencies but it does not yet provision Redis as-a-service
+
+**Declarative Workflows**
+
+Whilst it would have been possible to install the operator and it’s dependencies with `kubectl apply` commands targeting the cluster,
```

Comment:
```suggestion
Whilst it would have been possible to install the operator and it's dependencies with `kubectl apply` commands targeting the cluster,
```